core/src/main/scripts/importer/validateData.py

```diff
@@ -3336,16 +3514,90 @@ class GsvaPvalueValidator(GsvaWiseFileValidator):
     calculate a p-value for each GSVA score using a bootstrapping method.
     """
 
+    def __init__(self, *args, **kwargs):
+        super(GsvaPvalueValidator, self).__init__(*args, **kwargs)
+
     # Score must be between -0 and 1
     def checkValue(self, value, col_index):
         """Check a value in a sample column."""
         stripped_value = float(value.strip())
         if stripped_value <= 0 or stripped_value > 1:
-            self.logger.error("Value is not between 0 and 1, and therefor not a valid p-value",
+            self.logger.error("Value is not between 0 and 1, and therefore not a valid p-value",
                               extra={'line_number': self.line_number,
                                      'column_number': col_index + 1,
                                      'cause': value})
 
+
+
+class TreatmentValidator(TreatmentWiseFileValidator):
+
+    """ Validator for files containing treatment response values.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(TreatmentValidator, self).__init__(*args, **kwargs)
+
+    # (1) Natural positive number (not 0)
+    # (2) Number may be prefixed by ">" or "<"; f.i. ">n" means that the treatment was ineffective at the highest tested concentration of n.
+    # (3) NA cell value is allowed; means treatment was not tested on a sample
+    # (4) Is an empty cell value allowed? (meaning treatment was not tested on a sample)
+    #
+    # Warnings for concentrations:
+    # (1) Cell contains a value without decimals and is not prependend by ">"; value appears to be truncated but lacks ">" truncation indicator
+    def checkValue(self, value, col_index):
+        """Check a value in a sample column."""
+
+        # value is not defined
+        if value == "":
+            self.logger.error("Cell is empty. An effective concentration value is expected. Use 'NA' to indicate missing values.",
+                extra={'line_number': self.line_number,
+                'column_number': col_index + 1,
+                'cause': value})
+            return
+        
+        stripped_value = value.strip()
+
+        # 'NA' is an allowed value. No further validations apply.
+        if stripped_value == 'NA':
+            return
+
+        # if the value is prefixed with '>' or '<' remove this prefix
+        # prior to evaluation of the numeric value
+        hasTruncSymbol = re.match("^[><]", stripped_value)
+        stripped_value = re.sub("^[><]","", stripped_value)
```

Comment:
what happens if values is like `> 10` instead of `>10`?

Reply:
They are not recognized. It is not mentioned that a space is allowed in File-Formats.md. But then again there is also no statement that it is not. As on the effect of my code. After removal of `>` or `<` the remaining string is passed through `float(stripped_value)`. I tried float(" 5") (notice the space here) and this evaluates to 5.0. So, if there would be a space no error would be raised and the value is accepted by the validator. I guess we may add a remark to the docs that a whitespace character is allowed (and adjust the study_es_0 to include one of these values)?

Reply:
maybe try `re.sub("^[><]\s*"`

Reply:
Just to be sure, it might than also be worth [prefixing the string with an `r`]( https://docs.python.org/3/tutorial/introduction.html#strings ), to mark that backslashes are not to be read as special characters in the string literal, but passed down verbatim. That's [generally recommended in regexes](https://docs.python.org/3.5/library/re.html#module-re).

Reply:
@fedde-s 
But the special characters of `"^[><]\s*" should be interpreted as special characters, not?

The regex was updated to `"^[><]\s*"`.

Reply:
Looks good. Only the quotes should be special characters as far as the string literal is concerned, before the string literal is passed down to be parsed by the `re.sub()` function. Without the `r"`, the reader would have to think about whether `\s` means anything special in Python string literal syntax, in addition to what the resulting string means as a regex.

Reply:
Don't forget to push the commit you just mentioned, though.